server/e2e/jobs/specs/library.e2e-spec.ts

```diff
@@ -236,6 +165,185 @@ describe(`${LibraryController.name} (e2e)`, () => {
         );
       });
     });
+
+    it('should offline a missing file', async () => {
+      await fs.promises.cp(`${IMMICH_TEST_ASSET_PATH}/albums/nature`, `${IMMICH_TEST_ASSET_TEMP_PATH}/albums/nature`, {
+        recursive: true,
+      });
+
+      const library = await api.libraryApi.create(server, admin.accessToken, {
+        type: LibraryType.EXTERNAL,
+        ownerId: admin.userId,
+        importPaths: [`${IMMICH_TEST_ASSET_TEMP_PATH}`],
+      });
+
+      await api.libraryApi.scanLibrary(server, admin.accessToken, library.id);
+
+      const onlineAssets = await api.assetApi.getAllAssets(server, admin.accessToken);
+      expect(onlineAssets.length).toBeGreaterThan(1);
+
+      await fs.promises.rm(`${IMMICH_TEST_ASSET_TEMP_PATH}/albums/nature/silver_fir.jpg`);
+
+      await api.libraryApi.scanLibrary(server, admin.accessToken, library.id);
+
+      const assets = await api.assetApi.getAllAssets(server, admin.accessToken);
```

Comment:
This is bound to break, correct? You have a race condition between the library scanning and the fetch for all assets.

Reply:
I don't see how this would race. In these e2e jobs, we mock the job queue so all queued jobs finish with the await

Reply:
Oh I see. Sorry I mixed that up then... Then it'll only become an issue once we move them to e2e/

Reply:
I'm not looking forward to that 😂